### ISSUE TITLE
TimeZoneInfo.FindSystemTimeZoneById("Asia/Tehran") on Debian GNU/Linux 9 returns null for AdjustmentRules, targeting .net core 2.0

### ISSUE THREAD

**Issue by ioanaromcea** [dotnet/corefx#29688]
`TimeZoneInfo.FindSystemTimeZoneById("Asia/Tehran");` on Debian GNU/Linux 9 returns null for AdjustmentRules, targeting .net core 2.0

Bellow you can find some details ..

Tried to get TimeZoneInfo for Iran both on Windows 10 and Debian GNU/Linux 9.

executing `TimeZoneInfo.FindSystemTimeZoneById("Iran Standard Time");`  on Windows, the AdjustmentRules field of TimeZoneInfo object that I get as result, contains an array of AdjustmentRule objects.
executing `TimeZoneInfo.FindSystemTimeZoneById("Asia/Tehran"); `on Linux, the AdjustmentRules field is null.

checked the content of time zone files from /usr/share/zoneinfo/ on my Linux system and there are some adjustment rules defined for Asia/Tehran in the corresponding file.
also, tried to change my local timezone to Asia/Tehran and the Date and time on my system was automatically adjusted at Iran Daylight Time.

also, tried to get the TimeZoneInfo for other regions like Europe/Bucharest, Australia/Sydney, Europe/Paris, Asia/Beirut, etc. and Asia/Tehran was the only one for which I encountered this issue.

My question is why AdjustmentRules for Asia/Tehran is null on Linux? Shouldn't it contain a list of adjustment rules too ?

**Comment by tarekgh** [comment]
@ioanaromcea we have fixed this issue in 2.1.  Is it possible you move to 2.1 instead?

**Comment by tarekgh** [comment]
https://github.com/dotnet/coreclr/issues/17393

**Comment by joperezr** [comment]
@tarekgh are we planning to service this fix on previous versions of .NET Core? If not, then I guess we should just close this issue as fixed.

**Comment by tarekgh** [comment]
@joperezr 

> are we planning to service this fix on previous versions of .NET Core? If not, then I guess we should just close this issue as fixed.

I am waiting @ioanaromcea response here to know if this is a blocking scenario and if they can move to 2.1 instead. then we can decide. but so far, there is no plan to port the fix to 2.0

**Comment by ioanaromcea** [comment]
@tarekgh is there any docker image that supports 2.1 version? Cause that would be our limitation in upgrading from 2.0 to 2.1

**Comment by joperezr** [comment]
Yes, we have a bunch of docker images for different distros that already have 2.1 RC1 on them. Here is a link where you can find the tags: https://store.docker.com/images/dotnet
